src/buildHTML.js

```diff
@@ -21,27 +21,48 @@ var isSpace = function(node) {
     return node instanceof domTree.span && node.classes[0] === "mspace";
 };
 
+// Binary atoms (first class `mbin`) change into ordinary atoms (`mord`)
+// depending on their surroundings. See TeXbook pg. 442-446, Rules 5 and 6,
+// and the text before Rule 19.
+
+var isBin = function(node) {
+    return node && node.classes[0] === "mbin";
+};
+
+var isBinLeftCanceller = function(node, isComplete) {
+    if (node) {
+        return utils.contains(["mbin", "mopen", "mrel", "mop", "mpunct"],
+                              node.classes[0]);
+    } else {
+        return isComplete;
+    }
+};
+
+var isBinRightCanceller = function(node, isComplete) {
+    if (node) {
+        return utils.contains(["mrel", "mclose", "mpunct"], node.classes[0]);
+    } else {
+        return isComplete;
+    }
+};
+
 /**
  * Take a list of nodes, build them in order, and return a list of the built
- * nodes. This function handles the `prev` node correctly, and passes the
- * previous element from the list as the prev of the next element, ignoring
- * spaces. documentFragments are flattened into their contents, so the
- * returned list contains no fragments.
+ * nodes. documentFragments are flattened into their contents, so the
+ * returned list contains no fragments. `isComplete` is true if `expression`
+ * is complete (no atoms will be added on either side).
```

Comment:
Maybe `isRealGroup` would be a better name than `isComplete`.  The `groupType` handlers for things like `color`, `styling`, etc. where `isComplete` is `false` are "groups" that disappear after this processing.